examples/example_multi_genome.py

```diff
@@ -0,0 +1,140 @@
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.constants
+from docopt import docopt
+
+import cgp
+
+args = docopt(docopt_str)
+
+# %%
+# We first define a target function.  The function applies different
+# transformations to the input depending whether the input is less or greater
+# than zero. We thus need to fit two different functions.
+
+
+def f_target(x):
+    return np.select([x < 0, x >= 0], [-x, x ** 2 + 1.0])
+
+
+# %%
+# Then we define the objective function for the evolution. It uses the
+# mean-squared error between the output of the expression represented by a given
+# individual and the target function evaluated on a set of random points. We
+# here either evaluate the function represented by the first (``f[0]``) or the second
+# genome (``f[1]``), depending whether the input is less or greater than zero.
+
+
+def objective(individual):
+    if individual.fitness is not None:
+        return individual
+
+    n_function_evaluations = 1000
+
+    np.random.seed(1234)
+
+    # Note that f is now a list of functions because individual is an instance
+    # of `InvidividualMultiGenome`
+    f = individual.to_numpy()
+    x = np.random.uniform(-4, 4, (n_function_evaluations, 1))
+    y = np.piecewise(x, [x[:, 0] < 0, x[:, 0] >= 0], f)[:, 0]
+    loss = np.sum((f_target(x[:, 0]) - y) ** 2)
+    individual.fitness = -loss / n_function_evaluations
+    return individual
+
+
+# %%
+# Next, we set up the evolutionary search. First, we define the parameters for
+# the population, the genomes of individuals, and the evolutionary
+# algorithm. Note that we define ``genome_params`` as a list of parameter
+# dictionaries which causes the population to create instances of
+# ``InvidividualMultiGenome``.
+
+population_params = {"n_parents": 1, "mutation_rate": 0.03, "seed": 8188211}
+
+single_genome_params = {
+    "n_inputs": 1,
+    "n_outputs": 1,
+    "n_columns": 12,
+    "n_rows": 1,
+    "levels_back": 5,
+    "primitives": (cgp.Add, cgp.Sub, cgp.Mul, cgp.ConstantFloat),
+}
+genome_params = [single_genome_params, single_genome_params]
+
+ea_params = {"n_offsprings": 4, "tournament_size": 2, "n_processes": 1}
+
+evolve_params = {"max_generations": int(args["--max-generations"]), "min_fitness": 0.0}
+
+# %%
+# We create a population that will be evolved
+pop = cgp.Population(**population_params, genome_params=genome_params)
+
+# %%
+# and an instance of the (mu + lambda) evolutionary algorithm
+ea = cgp.ea.MuPlusLambda(**ea_params)
+
+# %%
+# We define a callback for recording of fitness over generations
+history = {}
+history["fitness_champion"] = []
+
+
+def recording_callback(pop):
+    history["fitness_champion"].append(pop.champion.fitness)
+
+
+# %%
+# and finally perform the evolution
+cgp.evolve(pop, objective, ea, **evolve_params, print_progress=True, callback=recording_callback)
+
+
+# %%
+# After finishing the evolution, we plot the result.
```

Comment:
maybe it would be nice here to print the expressions encoded in the individual?

Reply:
yes, thanks, added it.